_P248_TempHumidity_AHT1x.ino

```diff
@@ -0,0 +1,339 @@
+#define AHT10_POWER_ON_DELAY       40    //at least 20..40 milliseconds
+#define AHT10_CMD_DELAY            350   //at least 300 milliseconds, no info in datasheet!!!
+#define AHT10_SOFT_RESET_DELAY     20    //less than 20 milliseconds
+
+#define AHT10_FORCE_READ_DATA      true  //force to read data
+#define AHT10_USE_READ_DATA        false //force to use data from previous read
+#define AHT10_ERROR                0xFF  //returns 255, if communication error is occurred
+
+ 
+#define AHTX0_I2CADDR_DEFAULT 0x38   ///< AHT default i2c address
+#define AHTX0_CMD_CALIBRATE 0xE1     ///< Calibration command
+#define AHTX0_CMD_TRIGGER 0xAC       ///< Trigger reading command
+#define AHTX0_CMD_SOFTRESET 0xBA     ///< Soft reset command
+#define AHTX0_STATUS_BUSY 0x80       ///< Status bit for busy
+#define AHTX0_STATUS_CALIBRATED 0x08 ///< Status bit for calibrated
+
+
+#ifdef USE_AHT2x
+  #define AHTX_CMD     0xB1 // Cmd for AHT2x
+#else
+  #define AHTX_CMD     0xE1 // Cmd for AHT1x
+#endif
+
+uint8_t AHTSetCalCmd[3]    = { AHTX_CMD, 0x08, 0x00 }; // load factory calibration coeff
+uint8_t AHTSetCycleCmd[3]  = { AHTX_CMD, 0x28, 0x00 }; // enable cycle mode
+uint8_t AHTMeasureCmd[3]   = { 0xAC, 0x33, 0x00 };     // start measurment command
+uint8_t AHTResetCmd        =   0xBA;                   // soft reset command
+
+uint8_t          AHT10_rawDataBuffer[6] = {AHT10_ERROR, 0, 0, 0, 0, 0};
+
+
+bool AHT1XStartMeasurement(uint8_t address) {
+  Wire.beginTransmission(address);
+  Wire.write(AHTMeasureCmd, 3);
+  if (Wire.endTransmission() != 0)
+    return false;
+  return true;
+}
+
+bool AHT1XReadMesurement(uint8_t address) {
+  
+  Wire.requestFrom(address, (uint8_t) 6);
+
+  for(uint8_t i = 0; Wire.available() > 0; i++) {
+     AHT10_rawDataBuffer[i] = Wire.read();
+  }
+  if (AHT10_rawDataBuffer[0] & 0x80)
+    return false; //device is busy
+
+  //humidity    = (((data[1] << 12)| (data[2] << 4) | data[3] >> 4) * AHT_HUMIDITY_CONST / KILOBYTE_CONST);
+  //temperature = ((AHT_TEMPERATURE_CONST * (((data[3] & 0x0F) << 16) | (data[4] << 8) | data[5]) / KILOBYTE_CONST) - AHT_TEMPERATURE_OFFSET);
+
+  return true;
+  //return (!isnan(aht1x_sensors[aht1x_idx].temperature) && !isnan(aht1x_sensors[aht1x_idx].humidity) && (aht1x_sensors[aht1x_idx].humidity != 0));
+}
+
+
+/**************************************************************************/
+/*
+    readTemperature()
+    Read temperature, °C 
+    NOTE:
+    - temperature range      -40°C..+80°C
+    - temperature resolution 0.01°C
+    - temperature accuracy   ±0.3°C
+*/
+/**************************************************************************/
+float AHT10_getTemperature()
+{
+  if (AHT10_rawDataBuffer[0] == AHT10_ERROR) return AHT10_ERROR; //error handler, collision on I2C bus
+
+
+
+  uint32_t temperature = ((uint32_t)(AHT10_rawDataBuffer[3] & 0x0F) << 16) | ((uint16_t)AHT10_rawDataBuffer[4] << 8) | AHT10_rawDataBuffer[5]; //20-bit raw temperature data
```

Comment:
Not sure if this is always the intended result.
If you're casting anyway, why not cast it to `uint32_t` for each part when OR'ing it.

Reply:
I can do that. Was a cut/paste issue ;)